server/src/main/java/org/elasticsearch/rest/RestRequest.java

```diff
@@ -320,17 +337,38 @@ public HttpBody.Stream contentStream() {
     }
 
     /**
-     * @return content of the request body or throw an exception if the body or content type is missing
+     * Release underlying HTTP request and related buffers.
      */
-    public final BytesReference requiredContent() {
+    void close() {
```

Comment:
Maybe `releaseRequestBody` rather than the overly-generic `close()`?

Reply:
how about `request.body().close()`?